File: lib/Binary.py
from math import floor

class Binary2:
    def __init__(self, array):
        self.list = array
        self.list.sort()
        print(f"Sorted list: {self.list}")
        self.length = len(array)
    
    def translateToAscii(self):
        tempItem = ""
        tempList = []
        for i in self.list:
            for j in i:
                tempItem += str(ord(j))
            tempList.append(int(tempItem))
            tempItem = ''
        return tempList
    

    def search(self, target):
        array = self.translateToAscii()
        T = ""
        for i in target:
            T += str(ord(i))
        L = 0
        R = self.length - 1
        while L <= R:
            M = floor((L+R)/2)
            if array[M] < int(T):
                L = M + 1
            elif array[M] > int(T):
                R = M - 1
            else:
                return M
        return f'[{target}] Doesn\'t exist in the given list'

File: lib/binary.js
class Binary {
  constructor(array) {
    this.list = array;
    this.list.sort(function (a, b) {
      return b - a;
    });
    this.list.reverse();
    console.log("Sorted list: " + this.list);
    this.length = this.list.length;
  }

  search(target) {
    let left = 0;
    let right = this.length - 1;
    //let middle;

    while (left <= right) {
      let middle = Math.floor((left + right) / 2);
      if (this.list[middle] < target) {
        left = middle + 1;
      } else if (this.list[middle] > target) {
        right = middle - 1;
      } else {
        return middle;
      }
    }
    return target + " Doesn't exist in the given list";
  }
}

module.exports = Binary;
